src/modules/sync/types/fileContentModel.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  ContentTypeContracts,
+  ContentTypeSnippetContracts,
+  ElementContracts,
+  TaxonomyContracts,
+} from "@kontent-ai/management-sdk";
+
+import { Replace } from "../../../utils/types.js";
+
+export type TaxonomySyncModel =
+  & Omit<TaxonomyContracts.ITaxonomyContract, "id" | "last_modified" | "codename" | "terms">
+  & Readonly<{
+    codename: string;
+    terms: ReadonlyArray<TaxonomySyncModel>;
+  }>;
+
+export type ContentTypeSnippetsSyncModel =
+  & Omit<ContentTypeSnippetContracts.IContentTypeSnippetContract, "id" | "codename" | "last_modified" | "elements">
+  & Readonly<{
+    codename: string;
+    elements: ReadonlyArray<Replace<ElementContracts.IContentTypeElementContract, "codename", string>>;
```

Comment:
What about references inside specific elements (e.g. `allowed_content_types` in a linked items element)?

Reply:
well thats a good problem :D `ElementContracts.IContentTypeElementContract` is a type which does contain some "shared" properties (it also contains name which guidelines do not have). however fo example `IAssetElement` does not extend `ElementContracts.IContentTypeElementContract` this (that's why when are doing as unknown as. The types are not so great, so I don't know if that would even help. The solution would be to make type union of all elements but that should be done in SDK in the first place, so for me it does not make sense write it here, when we should change the SDK when the time comes. However, we would still need a function to replace codename? to codename. But i don't see a point doing it now. Or you have some better idea in mind?

Reply:
Sure, we can create the transformed types of different elements in the tasks that will implement the transformation.